packages/s3/lib/client.js

```diff
@@ -0,0 +1,41 @@
+// @ts-check
+import { S3Client } from '@aws-sdk/client-s3'
+
+/**
+ * Create a new S3 client.
+ *
+ * @param {import('@aws-sdk/client-s3').S3ClientConfigType} config S3 client configuration object.
+ * @returns {import('@aws-sdk/client-s3').S3Client} S3 client.
+ */
+export const newClient = (config) => {
+  const client = new S3Client(config)
+  return client
+}
+
+/**
+ * Generate a S3 client configuration object.
+ *
+ * @param {import('@aws-sdk/client-s3').S3ClientConfigType & {
+ *  accessKeyId?: string;
+ *  secretAccessKey?: string;
+ * }} params Parameters.
+ * @returns {import('@aws-sdk/client-s3').S3ClientConfigType} S3 client configuration object.
+ */
+export const generateConfig = ({ accessKeyId, secretAccessKey, ...s3Config }) => {
+  const config = { ...s3Config }
+
+  // set credentials if provided
+  if (accessKeyId && secretAccessKey) {
+    config.credentials = {
+      accessKeyId,
+      secretAccessKey,
+    }
+  }
+
+  // set a default region if none is provided
+  if (!s3Config.region) {
+    config.region = 'us-east-1'
```

Comment:
Why us-east-1 in particular?

Reply:
It's the default region that is used in general if no one is specified.
But for most cases, the region is not required.

Maybe I can use the `process.env.AWS_DEFAULT_REGION || 'us-east-1'` value, so that it can support AWS directly out of the box (some tools supports using the [environment variables from AWS](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html)).

Reply:
Not a bad idea if that's natural to S3 users